test/Microsoft.Kusto.ServiceLayer.UnitTests/ObjectExplorer/SmoModel/ServerNodeTests.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Kusto.ServiceLayer.Connection.Contracts;
+using Microsoft.Kusto.ServiceLayer.DataSource;
+using Microsoft.Kusto.ServiceLayer.DataSource.Metadata;
+using Microsoft.Kusto.ServiceLayer.ObjectExplorer.DataSourceModel;
+using Microsoft.Kusto.ServiceLayer.ObjectExplorer.Nodes;
+using Microsoft.Kusto.ServiceLayer.Utility;
+using Microsoft.SqlTools.Extensibility;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.Kusto.ServiceLayer.UnitTests.ObjectExplorer.SmoModel
```

Comment:
It's interesting that we still have a SmoModel. This is very Sql specific. If we need this interface, probably needs to be renamed to DataSourceModel?